pylib/Tools/Executor/sequential.py

```diff
@@ -292,14 +292,16 @@ def execute_sections(self, testDef):
                         parent = None
                     if parent:
                         parent_log = testDef.logger.getLog(parent)
-                        parent_loc = parent_log['location']
+                        if 'location' in parent_log:
+                            parent_loc = parent_log['location']
                     if checking_run_if:
                         testDef.logger.verbose_print("run_if command: %s" % run_if_cmd)
                         if run_if_cmd:
                             cmdargs = shlex.split(run_if_cmd)
                             if parent:
                                 original_loc = os.getcwd()
-                                os.chdir(parent_loc)
+                                if parent_loc is not None:
```

Comment:
parent_loc would not be defined. this will throw an error.

an example: https://www.online-python.com/PTYOAEbRrg

Reply:
nice catch...

Reply:
@ribab check now

Reply:
I think that fixes it